src/Framework/MockObject/MockBuilder.php

```diff
@@ -178,6 +178,34 @@ public function setMethods(array $methods = null): self
         return $this;
     }
 
+    /**
+     * Specifies the subset of methods to mock, requiring each to exist in the class
+     *
+     * @param string[] $methods
+     *
+     * @throws RuntimeException
+     */
+    public function setRealMethods(array $methods): self
+    {
+        $reflection = new \ReflectionClass($this->type);
+
+        foreach ($methods as $method) {
+            if (!$reflection->hasMethod($method)) {
+                throw new RuntimeException(
```

Comment:
Do we need a leading slash before `RuntimeExeception` here (and in the `@throws` annotation)?

Reply:
And we should have unit tests for the error case.

Reply:
I don't think the leading slash is needed in this case? It seems to be resolved correctly.

There is a test for the exception, but do you mean for the message itself?

Reply:
I don't get why `RuntimeException` gets resolved correctly, though - it's not `use`d, and this class is within the `PHPUnit\Framework\MockObject` namespace. Do you have an idea?

Reply:
Oh, right, now I see the test.

Reply:
They are both in PHPUnit\Framework\MockObject namespace though right? So i think that would be why.

Reply:
Ah, you're right. There indeed is a `PHPUnit\Framework\MockObject\RuntimeException` class. How confusing.

Reply:
In other words: This is not what I would have expected from the code, and the (existing) code does not follow the "minimize the number of surprises" principle here.